### PR TITLE
usb: stm32: fix FIFO allocation sizes

### LINKED ISSUE #70789: usb: writing to CDC-ACM interface other than first one crashes USB stack on stm32f767
**Describe the bug**
If two composite CDC-ACM interfaces are configured (e.g. in the `cdc_acm_composite` sample), writing data to the second interface (using e.g. `uart_poll_out()`) crashes the USB stack to the point of becoming completely unusable.

**To Reproduce**

1. `west build -b bare_test samples/subsys/usb/cdc_acm_composite/`
    - I don't have an stm32f767 devboard on hand yet, so I've only tested this on custom hardware with a very bare devicetree. I'm expecting this to work with `-b nucleo_f767zi` as well though, will update once the board arrives.
3. `west flash`
4. Observe serial devices showing up in dmesg
5. Open terminals on both interfaces (`/dev/serial/by-id/usbZEPHYR_Zephyr_CDC_ACM_Composite_sample_*-if00` and `-if02`)
6. Type something into the second terminal (`-if02`): it shows up in the first terminal
7. Type something into the first terminal: (`-if00`): dmesg reports `usb usb1-port11: disabled by hub (EMI?), re-enabling...`, serial interfaces disappear, don't reappear until zephyr board is reset

The reproduction is a little fragile sometimes, so try again if it doesn't work on the first attempt.

I have attempted to reproduce this with an STM32F769I DISCOVERY board (slightly MCU) with no success, but this board only exposes the USB-HS peripheral, not the affected USB-FS peripheral.

**Expected behavior**
Writing data to a CDC-ACM interface doesn't hang up the USB stack.

**Impact**
This makes using multiple USB serial interfaces impossible on affected devices.

**Logs and console output**

```
[00:00:00.000,000] <dbg> usb_cdc_acm: cdc_acm_init: Device dev 0x80098ac dev_data 0x200200b0 cfg 0x2002044c added to devlist 0x20020dc8
[00:00:00.000,000] <dbg> usb_cdc_acm: cdc_acm_init: Device dev 0x8009898 dev_data 0x20020000 cfg 0x20020470 added to devlist 0x20020dc8
*** Booting Zephyr OS build v3.6.0-1761-gdd0923d7ad8a ***
[00:00:00.001,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x2002044c status 2
[00:00:00.001,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device connected
[00:00:00.001,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x20020470 status 2
[00:00:00.001,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device connected
[00:00:00.001,000] <inf> cdc_acm_composite: Wait for DTR
[00:00:00.007,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x2002044c status 5
[00:00:00.007,000] <inf> usb_cdc_acm: Device suspended
[00:00:00.007,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x20020470 status 5
[00:00:00.007,000] <inf> usb_cdc_acm: Device suspended
[00:00:00.201,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x2002044c status 1
[00:00:00.201,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device reset detected
[00:00:00.201,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x20020470 status 1
[00:00:00.201,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device reset detected
[00:00:00.333,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x2002044c status 1
[00:00:00.333,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device reset detected
[00:00:00.333,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x20020470 status 1
[00:00:00.333,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device reset detected
[00:00:00.437,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x2002044c status 3
[00:00:00.437,000] <inf> usb_cdc_acm: Device configured
[00:00:00.437,000] <dbg> usb_cdc_acm: cdc_acm_read_cb: ep 1 size 0 dev_data 0x200200b0 rx_ringbuf space 512
[00:00:00.437,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x20020470 status 3
[00:00:00.437,000] <inf> usb_cdc_acm: Device configured
[00:00:00.437,000] <dbg> usb_cdc_acm: cdc_acm_read_cb: ep 2 size 0 dev_data 0x20020000 rx_ringbuf space 512
[00:00:00.437,000] <dbg> usb_cdc_acm: cdc_acm_class_handle_req: CDC_SET_LINE_CODING 9600 0 0 8
[00:00:00.439,000] <dbg> usb_cdc_acm: cdc_acm_class_handle_req: CDC_SET_LINE_CODING 9600 0 0 8
[00:00:26.147,000] <dbg> usb_cdc_acm: cdc_acm_write_cb: ep 82: written 0 bytes dev_data 0x200200b0
[00:00:26.147,000] <dbg> usb_cdc_acm: cdc_acm_class_handle_req: CDC_SET_CONTROL_LINE_STATE 0x3
[00:00:26.147,000] <dbg> usb_cdc_acm: cdc_acm_class_handle_req: CDC_SET_LINE_CODING 115200 0 0 8
[00:00:26.247,000] <dbg> usb_cdc_acm: tx_work_handler: Nothing to send
[00:00:34.303,000] <dbg> usb_cdc_acm: cdc_acm_write_cb: ep 84: written 0 bytes dev_data 0x20020000
[00:00:34.303,000] <dbg> usb_cdc_acm: cdc_acm_class_handle_req: CDC_SET_CONTROL_LINE_STATE 0x3
[00:00:34.303,000] <dbg> usb_cdc_acm: cdc_acm_class_handle_req: CDC_SET_LINE_CODING 115200 0 0 8
[00:00:34.335,000] <inf> cdc_acm_composite: DTR set, start test
[00:00:34.336,000] <dbg> usb_cdc_acm: cdc_acm_int_in: CDC_IntIN EP[81]
[00:00:34.344,000] <dbg> usb_cdc_acm: cdc_acm_int_in: CDC_IntIN EP[81]
[00:00:34.403,000] <dbg> usb_cdc_acm: tx_work_handler: Nothing to send
[00:00:35.352,000] <dbg> usb_cdc_acm: cdc_acm_int_in: CDC_IntIN EP[83]
[00:00:35.360,000] <dbg> usb_cdc_acm: cdc_acm_int_in: CDC_IntIN EP[83]
[00:00:42.457,000] <dbg> usb_cdc_acm: cdc_acm_read_cb: ep 1 size 1 dev_data 0x200200b0 rx_ringbuf space 512
[00:00:42.457,000] <dbg> usb_cdc_acm: cdc_acm_fifo_read: dev 0x80098ac size 64 rx_ringbuf space 511
[00:00:42.457,000] <dbg> usb_cdc_acm: cdc_acm_fifo_fill: dev_data 0x20020000 len 1 tx_ringbuf space 512
[00:00:42.457,000] <dbg> usb_cdc_acm: tx_work_handler: Got 1 bytes from ringbuffer send to ep 84
[00:02:11.227,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x2002044c status 4
[00:02:11.227,000] <inf> usb_cdc_acm: Device disconnected
[00:02:11.227,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x20020470 status 4
[00:02:11.227,000] <inf> usb_cdc_acm: Device disconnected
[00:02:11.227,000] <dbg> usb_cdc_acm: cdc_acm_read_cb: ep 1 size 0 dev_data 0x200200b0 rx_ringbuf space 512
[00:02:11.227,000] <dbg> usb_cdc_acm: cdc_acm_write_cb: ep 84: written 1 bytes dev_data 0x20020000
[00:02:11.227,000] <dbg> usb_cdc_acm: cdc_acm_write_cb: tx_ringbuf is empty
[00:02:11.227,000] <dbg> usb_cdc_acm: cdc_acm_read_cb: ep 2 size 0 dev_data 0x20020000 rx_ringbuf space 512
[00:02:18.563,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x2002044c status 2
[00:02:18.563,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device connected
[00:02:18.563,000] <dbg> usb_cdc_acm: cdc_acm_dev_status_cb: cfg 0x20020470 status 2
[00:02:18.563,000] <dbg> usb_cdc_acm: cdc_acm_do_cb: Device connected
```

- 00:26: first terminal is opened on `-if00`
- 00:34: second terminal is opened on `-if02`
- 00:42: letter "a" is typed into `-if00`, which immediately kills the USB connection
- 02:11: the device is physically disconnected from and reconnected to the USB port, but it does not enumerate anymore.

Corresponding dmesg output, with timestamps synchronized:

```
[0.708383] usb 1-11: new full-speed USB device number 126 using xhci_hcd
[0.862380] usb 1-11: New USB device found, idVendor=2fe3, idProduct=0002, bcdDevice= 3.06
[0.862395] usb 1-11: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[0.862400] usb 1-11: Product: Zephyr CDC ACM Composite sample
[0.862404] usb 1-11: Manufacturer: ZEPHYR
[0.862408] usb 1-11: SerialNumber: 3236510700260023
[0.867851] cdc_acm 1-11:1.0: ttyACM0: USB ACM device
[0.869148] cdc_acm 1-11:1.2: ttyACM1: USB ACM device
[42.887281] usb usb1-port11: disabled by hub (EMI?), re-enabling...
[42.887299] usb 1-11: USB disconnect, device number 126
[43.384770] usb 1-11: new full-speed USB device number 127 using xhci_hcd
[59.000791] usb 1-11: device descriptor read/64, error -110
[74.748857] usb 1-11: new full-speed USB device number 3 using xhci_hcd
[90.233078] usb 1-11: device descriptor read/64, error -110
[105.853294] usb usb1-port11: attempt power cycle
[106.269127] usb 1-11: new low-speed USB device number 4 using xhci_hcd
[106.269618] usb 1-11: Device not responding to setup address.
[106.481757] usb 1-11: Device not responding to setup address.
[106.689145] usb 1-11: device not accepting address 4, error -71
[106.817106] usb 1-11: new low-speed USB device number 6 using xhci_hcd
[106.817574] usb 1-11: Device not responding to setup address.
[107.025527] usb 1-11: Device not responding to setup address.
[107.233144] usb 1-11: device not accepting address 6, error -71
[107.233188] usb usb1-port11: unable to enumerate USB device
```

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Toolchain: Zephyr SDK 0.16.5
 - Commit SHA or Version used:
   - latest `main` at time of posting: dd0923d7ad8ac1bac39bcbb55418ebb678591991
   - `v3.6.0`

### PR DESCRIPTION
**Pull request by arbrauns**
See commit messages.

Fixes #70789.

### CODE REVIEW

File: drivers/usb/device/usb_dc_stm32.c [hunk 1]
```diff
@@ -132,12 +132,13 @@ static const struct gpio_dt_spec ulpi_reset =
 
 /* We need a minimum size for RX FIFO */
 #define USB_FIFO_RX_MIN 160
```
Comment (arbrauns) [inline drivers/usb/device/usb_dc_stm32.c:134]: The original PR (#62530) doesn't explain where this number comes from, and doesn't clarify whether it is supposed to be bytes or words. @Desvauxm-st?

Comment (arbrauns) [reply drivers/usb/device/usb_dc_stm32.c]: Judging by #69897, it's probably supposed to be words, updated accordingly.

Comment (erwango) [reply drivers/usb/device/usb_dc_stm32.c]: FYI @Desvauxm-st is now enjoying his retirement. @marwaiehm-st will have a look.

Comment (marwaiehm-st) [reply drivers/usb/device/usb_dc_stm32.c]: @arbrauns yes it's supposed to be en words

**Review by erwango — APPROVED** [review]

**Review by marwaiehm-st — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]

**Pull request merged**